### PR TITLE
Treat queriers referencing certain catalogs as non-deterministic

### PR DESCRIPTION
**Pull request by caithagoras**
We're seeing verification failure due to query reading from MySQL tables. We need a way to allow Verifier to mark them as non-deterministic if rerunning the control query does not identify the query as non-deterministic.

```
== RELEASE NOTES ==

Verifier Changes
* Add support for specifying non-deterministic catalogs by configuration property ``determinism.non-determinism-catalogs``.
Queries explicitly referencing tables from those catalogs are treated as non-deterministic.
```

### CODE REVIEW

File: presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ChecksumResult.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
     private final Map<String, Object> checksums;
 
     @VisibleForTesting
```
Comment (mbasmanova) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ChecksumResult.java]: perhaps, remove @VisibleForTesting; this method is a public method now

**Comment by caithagoras** [comment]
@mbasmanova comments addressed for the config property.

Regarding the further optimization,
- I've now move the check to the beginning of determinism analysis, so that control query won't get rerun for determinism analysis in those cases.
- We still need to run the control query. If control query fails, we skip the entire verification.
- We can skip checksum at all, and I can do that in a separate PR.

**Comment by caithagoras** [comment]
Could you accept the PR if you have no additional comments? Thanks!

**Comment by mbasmanova** [comment]
@caithagoras Please update release notes to reflect recent changes before merging.

**Pull request merged**